crates/edr_napi/src/config.rs

```diff
@@ -207,25 +227,82 @@ impl TryFrom<MiningConfig> for edr_provider::MiningConfig {
     }
 }
 
-impl TryFrom<ProviderConfig> for edr_napi_core::provider::Config {
-    type Error = napi::Error;
+impl ObservabilityConfig {
+    /// Resolves the instance, converting it to a
+    /// [`edr_provider::observability::Config`].
+    pub fn resolve(self, env: &napi::Env) -> napi::Result<edr_provider::observability::Config> {
+        let on_collected_coverage_fn = self
+            .code_coverage
+            .map(
+                |code_coverage| -> napi::Result<Box<dyn SyncOnCollectedCoverageCallback>> {
+                    let mut on_collected_coverage_callback: ThreadsafeFunction<
+                        _,
+                        ErrorStrategy::Fatal,
+                    > = code_coverage
+                        .on_collected_coverage_callback
+                        .create_threadsafe_function(
+                            0,
+                            |ctx: ThreadSafeCallContext<HashSet<Bytes>>| {
+                                let hits = ctx
+                                    .env
+                                    .create_array_with_length(ctx.value.len())
+                                    .and_then(|mut hits| {
+                                        for (idx, hit) in ctx.value.into_iter().enumerate() {
+                                            ctx.env
+                                                .create_buffer_with_data(hit.to_vec())
+                                                .and_then(|hit| {
+                                                    hits.set_element(idx as u32, hit.into_raw())
+                                                })?;
+                                        }
+                                        Ok(hits)
+                                    })?;
```

Comment:
Is there a reason to prefer `and_then` over `?`? This seems nicer to me:

```suggestion
                                let mut hits = ctx.env.create_array_with_length(ctx.value.len())?;

                                for (idx, hit) in ctx.value.into_iter().enumerate() {
                                    let hit = ctx.env.create_buffer_with_data(hit.to_vec())?;
                                    hits.set_element(idx as u32, hit.into_raw())?;
                                }
```

Reply:
From a styling perspective it comes down to preference (imperative vs functional). On our team, we don't prescribe one style over the other. It's up to the implementer.